theano/gpuarray/extra_ops.py

```diff
@@ -229,26 +235,23 @@ def gpu_kernels(self, node, nodename):
         return kernels
 
     def c_code(self, node, nodename, inp, out, sub):
-        if node.inputs[0].type.context.kind != b'cuda':
-            raise NotImplementedError("cuda only")
-        x, = inp
-        z, = out
-        axis = self.axis if self.axis is not None else 0
-        fail = sub['fail']
-        ctx = sub['params']
-
-        code = """
-
+        return """
+        if (%(params)s->device_is_not_cuda) {
```

Comment:
That if should be in python and we should raise an error in Python. This will allow Theano to fall back to the Python code. This is the old way of doing think. Why did you changed it?